generateSettings.ts

```diff
@@ -215,4 +232,26 @@ text += ';\n';
 text += `export const settingsSchema = ${JSON.stringify(createSchema(true))};`;
 writeFileSync(`lib/settingsGenerated.ts`, text);
 
-writeFileSync(`settings.schema.json`, JSON.stringify(createSchema(false)));
\ No newline at end of file
+writeFileSync(`settings.schema.json`, JSON.stringify(createSchema(false)));
+
+let readmeText = readFileSync('README-input.md', {encoding: 'utf-8'}) + '\n';
+function ISchemaObject(schema: Schema): schema is ISchemaObject {
+  return schema.type === 'object' && Object.values(schema.properties).length > 0;
+}
+function schemaWalk(schema: Schema, path: string[]) {
+  if (ISchemaObject(schema)) {
+    if (path.length > 0) {
+      readmeText += '#'.repeat(path.length + 1) + ` ${path.join('.')}\n`;
+      readmeText += `|name|description|type|default|\n`;
+      readmeText += `|---|---|---|---|\n`;
+    }
+    for (const [key, value] of Object.entries(schema.properties)) {
+      schemaWalk(value, [...path, key]);
+    }
+  } else {
+    readmeText += `|${path.join('.')}|${schema.description}|${schema.type}|${JSON.stringify(schema.default) ?? ''}\n`;
+  }
+}
+schemaWalk(createSchema(false), []);
+
```

Comment:
Do we want a manual ordering of the settings? Currently the first setting is the trace which is probably the least important one. I would suggest the following group ordering:
1. `analysis`
2. `rules`
3. `style`
4. `paths`
5. `semanticTokens*` Move to one of the other groups or make a group for it?
6. `trace` Does this need to be a group/heading?

Reply:
Thinking about this. I realized we need to split the settings. Some of the stuff should clearly be in the configuration file (like style and stuff). But some of the settings are not project specific but clearly use preference and should definitly move back to the vscode settings: `semantikTokens`, `trace`.
Also wondering about additionalPathes. I currently have my personal absolute path of vunit in there.
My proposal:

- Move the settings definition out of package.json to its own schema file. And keep that as the original source.
- Move the stuff settings stuff back to package.json
- Make the naming clear: Eg. configuration for the stuff moving to the file settings for the stuff coming from the editor.
- Make a PR changing the above points. Afterwards continue with this?

Reply:
Yes, sounds good. I also don't know how the additional paths could be handled... Especially with #441 it doesn't make sense in the settings file for CI but I also don't want to put them into vs code.
Maybe we could support shell replacements and then use `$(poetry env info --path)/lib/python*/site-packages/vunit/vhdl/` but that seems really dangerous xD Are there any other linters that somehow handle this case for a language without a standard build process? C maybe?